caddyhttp/fastcgi/fastcgi.go

```diff
@@ -91,14 +124,18 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 				resp, err = fcgiBackend.Post(env, r.Method, r.Header.Get("Content-Type"), r.Body, contentLength)
 			}
 
-			if resp.Body != nil {
-				defer resp.Body.Close()
-			}
-
 			if err != nil && err != io.EOF {
+				persistentConnections[network+address].Close()
+				persistentConnections[network+address] = nil
```

Comment:
I though in go there is a garbage collector, i.e. no explicit deallocation? Deleting from the map is a good idea, anyway, that saves one comparison. 

My intention was to use 
```
poorMansSerialisation[network+address].Lock()
defer poorMansSerialisation[network+address].Unlock()
```
to lock the map until returning from ServeHTTP. Maybe I misunderstood the way mutex work in go?